fdbserver/LogRouter.actor.cpp

```diff
@@ -148,7 +153,12 @@ struct LogRouterData {
 		specialCounter(cc, "WaitForVersionMaxMS", [this](){ double val = this->maxWaitForVersionTime; this->maxWaitForVersionTime = 0; return 1000*val; });
 		specialCounter(cc, "GetMoreMS", [this](){ double val = this->getMoreTime; this->getMoreTime = 0; return 1000*val; });
 		specialCounter(cc, "GetMoreMaxMS", [this](){ double val = this->maxGetMoreTime; this->maxGetMoreTime = 0; return 1000*val; });
-		logger = traceCounters("LogRouterMetrics", dbgid, SERVER_KNOBS->WORKER_LOGGING_INTERVAL, &cc, "LogRouterMetrics");
+		specialCounter(cc, "Geneartion", [this]() { return this->generation; });
+		logger = traceCounters("LogRouterMetrics", dbgid, SERVER_KNOBS->WORKER_LOGGING_INTERVAL, &cc,
+		                       "LogRouterMetrics", [this](TraceEvent& te) {
```

Comment:
Why do we need this decorator pattern, instead of using two specialCounters for PrimaryPeekLocation and RouterTag respectively?

Reply:
`specialCounter()` requires the lambda to return `int64_t`. The change made in this PR would allow user to log non-integer information.

Reply:
One suggestion is to explore using decltype here. Something of this sort.

`auto getValue() const override -> decltype(f()) { return f(); }`

Reply:
It's a bit complex since the change would involve `ICounter` interface and its stringify function for `Traceable`. Plus, a `getValue()` call on a `Counter` returns non-numbers feels weird.